api/src/main/java/org/pmiops/workbench/api/RuntimeController.java

```diff
@@ -252,6 +253,13 @@ private Runtime getOverrideFromListRuntimes(String googleProject) {
       runtime = new Runtime();
     }
 
+    GceWithPdConfig gceWithPdConfig = runtime.getGceWithPdConfig();
+    if (gceWithPdConfig != null) {
+      PersistentDiskRequest persistentDiskRequest = gceWithPdConfig.getPersistentDisk();
+      if (persistentDiskRequest != null && persistentDiskRequest.getName().isEmpty()) {
```

Comment:
what happened when `!persistentDiskRequest.getName().isEmpty()` seems you will just use the name passed in the request. Is that expected? Another option is to always ignore the name in the request, and always override in the controller here.

Reply:
Here it's used to attach an existing PD to the new runtime

Reply:
Haven't seen UI code for this part yet. To confirm, in Ui's create runtime, 

1.  create runtime with an existing PD: gceWithPdConfig. getPersistentDisk is not null, and it has non empty name. 
2. create runtime with new PD: gceWithPdConfig. getPersistentDisk is not null but with empty name?

Reply:
Exactly. Specifically, for case 1, the UI will get the existing PD name by disk api first and then use this name in gceWithPdConfig. The UI code for this part hasn't been completed yet.